tools/workbench/src/main/java/org/eclipse/rdf4j/workbench/proxy/WorkbenchServlet.java

```diff
@@ -127,7 +134,27 @@ private void handleRequest(final HttpServletRequest req, final HttpServletRespon
 		} catch (UnauthorizedException e) {
 			handleUnauthorizedException(req, resp);
 		} catch (RepositoryConfigException | RepositoryException e) {
-			throw new ServletException(e);
+			if (e.getCause() instanceof RemoteShaclSailValidationException) {
+				Model model = ((RemoteShaclSailValidationException) e.getCause()).validationReportAsModel();
+
+				resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+				resp.setContentType(TEXT_PLAIN);
+				PrintWriter writer = resp.getWriter();
+
+				writer.println("SHACL validation failed with the following report:\n");
+				WriterConfig writerConfig = new WriterConfig();
+				writerConfig.set(BasicWriterSettings.PRETTY_PRINT, true);
+				writerConfig.set(BasicWriterSettings.INLINE_BLANK_NODES, true);
+				Rio.write(model, writer, RDFFormat.TURTLE, writerConfig);
+
+				writer.println(
+						"\n" +
+								"THIS ERROR MESSAGE IS EXPERIMENTAL AND IS SUBJECT TO CHANGE - " +
```

Comment:
Why are you sending the validation report with this tagged on, and as plain text? Wouldn't it make more sense to send just the turtle data, as text/turtle?

Reply:
It’s just a simple way to display it to the user. In the future someone could extend the user interface to show the error properly, at which point it could be changed to work some other way.

Reply:
To explain further. This is actually a view layer issue...we need to return this error to the view layer in some way, but I've not started looking at the view layer of the workbench to figure out how to interact with it. Returning the text/plain here is just bypassing the view layer, returning text/turtle is not correct because the view layer is supposed to be html. Text/plain is just a temporary hack to make the error visible to the user. And by temporary I mean, probably permanent unless someone who built the workbench can help out.

Reply:
I, um, did. But it was a while back. Let me have a think about this, I don't want to make REST API decisions based on shortcomings of the client tool.

Reply:
Hang on, apologies, I completely misread this - for some reason I assumed I was looking at the RDF4J Server code, not the workbench. So yes - this makes much more sense now!